src/rez/rex.py

```diff
@@ -1169,9 +1169,20 @@ def execute_code(self, code, filename=None):
             code (str): Rex code to execute.
             filename (str): Filename to report if there are syntax errors.
         """
-        self.compile_code(code=code,
-                          filename=filename,
-                          exec_namespace=self.globals)
+        # we want to execute the code using self.globals - if for no other
+        # reason that self.formatter is pointing at self.globals, so if we
+        # passed in a copy, we would also need to make self.formatter "look" at
+        # the same copy - but we don't want to "pollute" our namespace, because
+        # the same executor may be used to run multiple packages. Therefore,
+        # we save a copy of self.globals before execution, and restore it after
+        saved_globals = dict(self.globals)
+        try:
+            self.compile_code(code=code,
+                              filename=filename,
+                              exec_namespace=self.globals)
+        finally:
+            self.globals.clear()
+            self.globals.update(saved_globals)
```

Comment:
I'm gonna do this as a separate release anyway, since there's the outside chance that a studio is accidentally relying on this buggy behavior (and it'll be clearer to point that out in a separate release). Keen to get this done so I'll also do the self.globals.copy instead, happy to change back after if there's something I've missed.